File: ios/BlueViewWeather/Views/Components/CurrentWeatherCard.swift
import SwiftUI

struct CurrentWeatherCard: View {
    let city: String
    let current: CurrentDto
    let daily: DailyDto
    let units: String

    private var sym: String { units == "imperial" ? "°F" : "°C" }
    private var spdU: String { units == "imperial" ? "mph" : "km/h" }
    private var isDay: Bool { current.isDay == 1 }
    private var hiLo: String {
        guard !daily.time.isEmpty else { return "" }
        return "↑ \(Int(daily.tempMax[0]))  ↓ \(Int(daily.tempMin[0]))  \(sym)"
    }

    var body: some View {
        VStack(alignment: .leading, spacing: 12) {
            Text(city)
                .font(.title3)
                .foregroundColor(.blueAccent)

            HStack(alignment: .center, spacing: 16) {
                Text(wmoIcon(current.weatherCode, isDay: isDay))
                    .font(.system(size: 64))

                VStack(alignment: .leading, spacing: 2) {
                    Text("\(Int(current.temperature))\(sym)")
                        .font(.system(size: 44, weight: .regular))
                        .foregroundColor(.textPrimary)
                    Text(wmoDescription(current.weatherCode))
                        .font(.body)
                        .foregroundColor(.textSecondary)
                    if !hiLo.isEmpty {
                        Text(hiLo)
                            .font(.subheadline.weight(.semibold))
                            .foregroundColor(.tempHigh)
                    }
                }
            }

            HStack(alignment: .top, spacing: 16) {
                DetailChip("Feels \(Int(current.apparentTemperature))\(sym)")
                DetailChip("💧 \(current.humidity)%")
                DetailChip("💨 \(Int(current.windSpeed)) \(spdU) \(windDirLabel(current.windDirection))")
                DetailChip("👁 \(Int(current.visibility / 1000)) km")
            }
        }
        .padding(.horizontal, 20)
        .padding(.vertical, 16)
        .frame(maxWidth: .infinity, alignment: .leading)
    }
}

private struct DetailChip: View {
    let text: String
    init(_ text: String) { self.text = text }
    var body: some View {
        Text(text)
            .font(.subheadline)
            .foregroundColor(.textSecondary)
            .fixedSize(horizontal: false, vertical: true)
    }
}

File: app/src/main/java/ai/blueview/weather/ui/components/CurrentWeatherCard.kt
package ai.blueview.weather.ui.components

import androidx.compose.foundation.layout.*
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import ai.blueview.weather.data.api.dto.CurrentDto
import ai.blueview.weather.data.api.dto.DailyDto
import ai.blueview.weather.ui.theme.*
import ai.blueview.weather.util.*

@Composable
fun CurrentWeatherCard(
    city: String,
    current: CurrentDto,
    daily: DailyDto,
    units: String,
    modifier: Modifier = Modifier
) {
    val sym    = if (units == "imperial") "°F" else "°C"
    val spdU   = if (units == "imperial") "mph" else "km/h"
    val isDay  = current.isDay == 1
    val hiLo   = if (daily.time.isNotEmpty())
        "↑ ${daily.tempMax[0].toInt()}  ↓ ${daily.tempMin[0].toInt()}  $sym" else ""

    Column(modifier = modifier.padding(horizontal = 20.dp, vertical = 16.dp)) {
        // City
        Text(
            text  = city,
            style = MaterialTheme.typography.titleMedium,
            color = BlueAccent
        )
        Spacer(Modifier.height(12.dp))

        // Icon + Temperature row
        Row(verticalAlignment = Alignment.CenterVertically) {
            Text(
                text     = wmoIcon(current.weatherCode, isDay),
                fontSize = 64.sp
            )
            Spacer(Modifier.width(16.dp))
            Column {
                Text(
                    text  = "${current.temperature.toInt()}$sym",
                    style = MaterialTheme.typography.displayLarge,
                    color = TextPrimary
                )
                Text(
                    text  = wmoDescription(current.weatherCode),
                    style = MaterialTheme.typography.bodyLarge,
                    color = TextSecondary
                )
                if (hiLo.isNotBlank()) {
                    Text(
                        text  = hiLo,
                        style = MaterialTheme.typography.bodyMedium,
                        color = TempHigh,
                        fontWeight = FontWeight.SemiBold
                    )
                }
            }
        }

        Spacer(Modifier.height(12.dp))

        // Details row
        Row(
            horizontalArrangement = Arrangement.spacedBy(16.dp),
            modifier = Modifier.fillMaxWidth()
        ) {
            DetailChip("Feels ${current.apparentTemperature.toInt()}$sym")
            DetailChip("💧 ${current.humidity}%")
            DetailChip("💨 ${current.windSpeed.toInt()} $spdU ${windDirLabel(current.windDirection)}")
            DetailChip("👁 ${(current.visibility / 1000).toInt()} km")
        }
    }
}

@Composable
private fun DetailChip(text: String) {
    Text(text = text, style = MaterialTheme.typography.bodyMedium, color = TextSecondary)
}
